### PR TITLE
feat(stdlib): add `uuid_v7` function

### LINKED ISSUE #737: Add `uuid_v7` function
Due to the already existing `uuid_v4()` function for randomized UUID generation in format version 4, I would like to use the timestamp based UUID version 7 generation in vector-based `remap` types by just simply calling a VRL function `uuid_v7()` to generate a current UUID v7 identifier.

Moreover, the `uuid_v7()` function shall allow an optional parameter to pass a given timestamp (by default it corresponds to the `now()` function) to generate the randomized portion of the provided timestamp.

### PR DESCRIPTION
**Pull request by ppaulweber**
This PR introduces the following changes:

* provides new VRL function `uuid_v7()` with optional timestamp parameter
* adds unit as well as function-based source tests
* enables `v7` feature of the `uuid` dependency
* closes #737

### CODE REVIEW

**Review by pront — APPROVED** [review]
Thank you for this PR @ppaulweber. It looks good to me.

File: src/stdlib/uuid_v7.rs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        want: TypeDef::bytes().infallible(),
+    }];
+
+    #[test]
+    fn uuid_v7() {
+        let mut state = state::RuntimeState::default();
+        let mut object: Value = Value::Object(BTreeMap::new());
+        let tz = TimeZone::default();
+        let mut ctx = Context::new(&mut object, &mut state, &tz);
+        let value = UuidV7Fn { timestamp: None }.resolve(&mut ctx).unwrap();
```
[the remark this answers is not in the document]
Comment (ppaulweber) [reply src/stdlib/uuid_v7.rs:105]: @pront indeed, we could improve the internal test suite for this case as well. For now there is the `timestamp`-based use case provided in the function source-based tests in yet another nice ASCII art supported test case, see:

https://github.com/vectordotdev/vrl/blob/0da03256422dfe56ed8b7c7c89014a5e3c06a590/lib/tests/tests/functions/uuid_v7_valid_with_timestamp.vrl#L3-L9

**Comment by pront** [comment]
This needs a changelog fragment.

**Comment by ppaulweber** [comment]
@pront added the missing changelog fragment, see: https://github.com/vectordotdev/vrl/pull/738/commits/2a4bdc00ea1fc4812c2c0865bb95074ab150ae0c#diff-68a3c19f92521d6e91458f14318467adc8cae19012edcb4ad7766d1051b96917

**Comment by pront** [comment]
> @pront added the missing changelog fragment, see: [2a4bdc0#diff-68a3c19f92521d6e91458f14318467adc8cae19012edcb4ad7766d1051b96917](https://github.com/vectordotdev/vrl/commit/2a4bdc00ea1fc4812c2c0865bb95074ab150ae0c#diff-68a3c19f92521d6e91458f14318467adc8cae19012edcb4ad7766d1051b96917)

One last thing: https://github.com/vectordotdev/vrl/actions/runs/8257108198/job/22614222614?pr=738

File: changelog.d/738_uuid_v7.feature.md [hunk 2]
```diff
@@ -0,0 +1 @@
+Add a `uuid_v7` function that generates UUIDv7 timestamp-based uniquely identifiers.
```
Comment (pront) [inline changelog.d/738_uuid_v7.feature.md:1]:
```suggestion
Add a `uuid_v7` function that generates UUIDv7 timestamp-based unique identifiers.
```

**Comment by pront** [comment]
Changelog needs a fix: https://github.com/vectordotdev/vrl/actions/runs/8284476549/job/22670244934?pr=738

**Comment by ppaulweber** [comment]
> Changelog needs a fix: https://github.com/vectordotdev/vrl/actions/runs/8284476549/job/22670244934?pr=738

@pront thx, it just took me now a while what you are referring in this regard directly, because the provided example in the `changelog.d/README.md` is a bit misleading in that direction regarding the format, see: https://github.com/vectordotdev/vrl/blob/e5988a564023bd986f4bd45e6c5a64704525febc/changelog.d/README.md?plain=1#L71C1-L73

Would propose that maybe in another or follow-up PR to change the line maybe as suggested in the following:

```diff
- $ cat changelog.d/42_very_good_words.breaking.md
+ $ cat changelog.d/42.breaking.md
```

**Comment by pront** [comment]
> > Changelog needs a fix: https://github.com/vectordotdev/vrl/actions/runs/8284476549/job/22670244934?pr=738
> 
> @pront thx, it just took me now a while what you are referring in this regard directly, because the provided example in the `changelog.d/README.md` is a bit misleading in that direction regarding the format, see: https://github.com/vectordotdev/vrl/blob/e5988a564023bd986f4bd45e6c5a64704525febc/changelog.d/README.md?plain=1#L71C1-L73
> 
> Would propose that maybe in another or follow-up PR to change the line maybe as suggested in the following:
> 
> ```diff
> - $ cat changelog.d/42_very_good_words.breaking.md
> + $ cat changelog.d/42.breaking.md
> ```

FYI @neuronull

**Comment by jszwedko** [comment]
> > > Changelog needs a fix: https://github.com/vectordotdev/vrl/actions/runs/8284476549/job/22670244934?pr=738
> > 
> > 
> > @pront thx, it just took me now a while what you are referring in this regard directly, because the provided example in the `changelog.d/README.md` is a bit misleading in that direction regarding the format, see: https://github.com/vectordotdev/vrl/blob/e5988a564023bd986f4bd45e6c5a64704525febc/changelog.d/README.md?plain=1#L71C1-L73
> > Would propose that maybe in another or follow-up PR to change the line maybe as suggested in the following:
> > ```diff
> > - $ cat changelog.d/42_very_good_words.breaking.md
> > + $ cat changelog.d/42.breaking.md
> > ```
> 
> FYI @neuronull

Mmm, yeah, I see. Either the README needs to be updated or the script that runs in CI to check the filename format needs to be updated.

**Comment by jszwedko** [comment]
@ppaulweber there are a couple of more CI failures when you get a moment. I tried to fix them locally, but it looks like [I don't have permissions to push to this branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/committing-changes-to-a-pull-request-branch-created-from-a-fork).

**Comment by ppaulweber** [comment]
@jszwedko already fixed them, see: https://github.com/vectordotdev/vrl/pull/738/files#diff-2e9d962a08321605940b5a657135052fbcef87b5e360662bb527c96d9a615542L205

**Comment by jszwedko** [comment]
Perfect, thanks @ppaulweber !

**Comment by jszwedko** [comment]
Actually, I think you still need to regenerate the license inventory using: https://github.com/DataDog/rust-license-tool

`dd-rust-license-tool write`

**Comment by ppaulweber** [comment]
@jszwedko already done :smiley:

**Comment by neuronull** [comment]
> > > > Changelog needs a fix: https://github.com/vectordotdev/vrl/actions/runs/8284476549/job/22670244934?pr=738
> > > 
> > > 
> > > @pront thx, it just took me now a while what you are referring in this regard directly, because the provided example in the `changelog.d/README.md` is a bit misleading in that direction regarding the format, see: https://github.com/vectordotdev/vrl/blob/e5988a564023bd986f4bd45e6c5a64704525febc/changelog.d/README.md?plain=1#L71C1-L73
> > > Would propose that maybe in another or follow-up PR to change the line maybe as suggested in the following:
> > > ```diff
> > > - $ cat changelog.d/42_very_good_words.breaking.md
> > > + $ cat changelog.d/42.breaking.md
> > > ```
> > 
> > 
> > FYI @neuronull
> 
> Mmm, yeah, I see. Either the README needs to be updated or the script that runs in CI to check the filename format needs to be updated.

Thanks for flagging. The example fragment in the "Pull Requests" section is correct. But the example at the bottom of the readme needs to be updated.

https://github.com/vectordotdev/vrl/pull/747

**Pull request merged**